net/redisclient.go

```diff
@@ -226,11 +249,43 @@ func NewRedisRingClient(ro *RedisOptions) *RedisRingClient {
 		r.ring = redis.NewRing(ringOptions)
 		r.log = ro.Log
 		r.metricsPrefix = ro.MetricsPrefix
+
+		if ro.AddrUpdater != nil {
+			if ro.UpdateInterval == 0 {
+				ro.UpdateInterval = defaultUpdateInterval
+			}
+			go func() {
+				old := len(ringOptions.Addrs)
+				t := time.NewTicker(ro.UpdateInterval)
+				r.log.Info("Start goroutine to update redis instances every %s", ro.UpdateInterval)
+				for range t.C {
+					select {
+					case <-r.quit:
+						return
+					default:
+					}
+					addrs := ro.AddrUpdater()
+					if old != len(addrs) && len(addrs) != 0 {
+						r.log.Infof("Redis updater updating %d != %d", old, len(addrs))
+						r.SetAddrs(context.Background(), addrs)
+					}
+				}
+			}()
+		}
 	}
 
 	return r
 }
 
+func createAddressMap(addrs []string) map[string]string {
+	res := make(map[string]string)
+	for idx, addr := range addrs {
+		addr = strings.TrimPrefix(addr, "TCP://")
```

Comment:
I think this should be done inside of the `AddrUpdater` as this prefix is specific to the k8s endpoints.

Reply:
yes but we loop here and not there..

Reply:
fixed